mysql-test/main/information_schema_temp_table_lowercase.test

```diff
@@ -0,0 +1,14 @@
+--echo #
+--echo # MDEV-32973: SHOW TABLES LIKE shows temporary tables with
+--echo #             non-matching names
+--echo #
+
+# Compare table names in a case-insensitive manner
+create temporary table t1 (a int);
+select table_name from information_schema.tables where
+  table_schema='test' and table_name='T1';
```

Comment:
eh, really?
```
MariaDB [test]> create temporary table t1 (a int);
Query OK, 0 rows affected (0.002 sec)

MariaDB [test]> show create table T1;
ERROR 1146 (42S02): Table 'test.T1' doesn't exist
```
Why do you think table names should be compared case-insensitively?
See https://mariadb.com/kb/en/identifier-case-sensitivity/